libcontainer/nsenter/nsexec.c

```diff
@@ -699,8 +699,10 @@ void send_idmapsources(int sockfd, pid_t pid, char *idmap_src, int idmap_src_len
 					    AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT);
 		if (fd_tree < 0) {
 			sane_kill(pid, SIGKILL);
-			if (errno == EINVAL)
-				bail("failed to use open_tree(2) with path: %s, the kernel doesn't supports ID-mapped mounts", idmap_src);
+			if (errno == ENOSYS)
+				bail("failed to use open_tree(2), the kernel doesn't support ID-mapped mounts");
+			else if (errno == EINVAL)
+				bail("failed to use open_tree(2) with path: %s, the kernel doesn't support ID-mapped mounts", idmap_src);
```

Comment:
Maybe change that to `open_tree(2) failed`..., as the "use" is kind of implied here.

Reply:
Fixed, now errors are shorter. Thanks! :)